File: server/internal/model/preference.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Preference struct {
	ID        uint64         `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID    uint64         `gorm:"not null;uniqueIndex:idx_pref_user_key" json:"user_id"`
	Key       string         `gorm:"not null;size:100;uniqueIndex:idx_pref_user_key" json:"key"`
	Value     string         `gorm:"type:text" json:"value"`
	CreatedAt time.Time      `gorm:"not null" json:"created_at"`
	UpdatedAt time.Time      `gorm:"not null" json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

func (Preference) TableName() string { return "preferences" }

File: web/src/types/preference.ts
/**
 * 用户偏好相关类型定义
 * 对应后端 model/preference.go 和 DTO
 */

/**
 * 用户偏好数据类型
 * 存储用户个性化配置（键值对）
 */
export interface Preference {
  /** 偏好键名 */
  key: string;
  /** 偏好值 */
  value: string;
}

/**
 * 设置偏好请求参数
 */
export interface SetPreferenceRequest {
  /** 偏好键名 */
  key: string;
  /** 偏好值 */
  value: string;
}
